rll-examples/install-packages.sh

```diff
@@ -0,0 +1,114 @@
+#! /bin/bash
+# ---
+# RightScript Name: SYS packages install
+# Description: |
+#   Installs packages required by RightScripts.
+#   To handle naming variations, prefix package names with "yum:" or "apt:".
+# Inputs:
+#   PACKAGES_INSTALL:
+#     Input Type: single
+#     Category: System
+#     Description: |
+#       Set to "false" in order to skip combined package installation of RS_PACKAGES.
+#     Required: true
+#     Advanced: true
+#     Default: text:true
+#     Possible Values:
+#       - text:true
+#       - text:false
+# ...
+
+[ "$PACKAGES_INSTALL" == "false" ] && exit 0
+
+if which apt-get > /dev/null 2>&1
+then
+  pkgman=apt
+elif which yum > /dev/null 2>&1
+then
+  pkgman=yum
+fi
+
+if [ -z "$pkgman" -a -n "$RS_PACKAGES" ]
+then
+  echo "ERROR: Unrecognized package manager, but we have packages to install"
+  echo "To disregard this error, set PACKAGES_INSTALL to false"
+  exit 1
+else
+  echo "Detected package manager: $pkgman"
+fi
+
+# Determine which packages are suitable for install on this system
+declare -a list
+sz=0
+for pkg in $RS_PACKAGES
+do
+  echo $pkg | grep -Eq '^[a-z0-9_]+:'
+  selective=$?
```

Comment:
This script is not `-x` because of the amount of extraneous assignments and such, so there will be no 0/1 outputs.

I'm grabbing grep's return code here, not its output. Are you suggesting that I switch to capturing its output and testing for empty output, instead of just looking at its return code? Return code is cleaner, no?